langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/Filter.java

```diff
@@ -52,17 +53,19 @@
     public static BMap filter(BMap<?, ?> m, BFunctionPointer<Object, Boolean> func) {
         Type mapType = m.getType();
         Type newMapType;
+        Type newConstraint;
```

Comment:
Can't we directly create `newMapType` for the map and record cases instead of introducing a separate variable for the constraint?